conn.go

```diff
@@ -655,7 +657,7 @@ func (c *conn) connReader() {
 		// parse the frame
 		b, ok := buf.Next(bodySize)
 		if !ok {
-			c.connErr <- io.EOF
+			c.connErr <- fmt.Errorf("buffer EOF; requested bytes: %d, actual size: %d", bodySize, buf.Len())
```

Comment:
Should this be wrapped in a ConnectionError?

Reply:
In the mux, `c.connErr` is propagated to `c.err` before exiting, so it will get wrapped automatically by `conn.Err()`